### PR TITLE
Cnetextract App and Test Update

### PR DESCRIPTION
**Pull request by acpaquette**
App conversion for cnetextract and test conversion to gtests

## Description
Updates the cnetextract app format and converts all existing make tests into gtests.

## Related Issue
N/A

## Motivation and Context
For the push to convert apps and gtests

## How Has This Been Tested?
These are tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/FunctionalTestsCnetextract.cpp
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsCnetextract.cpp:31]: Not sure if worth the effort, but even with these simple failure tests, they tend to be separated into individual tests unless maybe one depends on the other (which, other than recycling inputs, they dont?).

**Review by AustinSanders — DISMISSED** [review]
Everything seems to check out here, but I have virtually no experience with cnetextract or control networks in general.  Might be worth waiting on another review.

File: isis/src/control/apps/cnetextract/cnetextract.h
Comment (Kelvinrr) [inline isis/src/control/apps/cnetextract/cnetextract.h:35]: missing extern, but now I'm thinking it's superfluous? I don't see a case where we plan to call the app without a header and it probably increases linking times 🤔

**Review by Kelvinrr — COMMENTED** [review]
just some stuff in line. looks good to me.

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**
